backend/test/v2/api/matcher/pipeline_matcher.go

```diff
@@ -0,0 +1,52 @@
+package matcher
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/kubeflow/pipelines/backend/test/v2/api/logger"
+	"github.com/onsi/ginkgo/v2"
+
+	model "github.com/kubeflow/pipelines/backend/api/v2beta1/go_http_client/pipeline_upload_model"
+	. "github.com/onsi/gomega"
+)
+
+func MatchPipelines(actual *model.V2beta1Pipeline, expected *model.V2beta1Pipeline, isKubeflowMode bool) {
+	Expect(actual.PipelineID).To(Not(BeEmpty()), "Pipeline ID is empty")
```

Comment:
For all these assertions, please use `ExpectWithOffset` so that the assertion failure is reported on the line that called `MatchPipelines` rather than this line here. This applies to other custom matchers.